packages/@aws-cdk/core/lib/private/cloudformation-lang.ts

```diff
@@ -204,3 +337,64 @@ export function isNameOfCloudFormationIntrinsic(name: string): boolean {
   // these are 'fake' intrinsics, only usable inside the parameter overrides of a CFN CodePipeline Action
   return name !== 'Fn::GetArtifactAtt' && name !== 'Fn::GetParam';
 }
+
+/**
+ * Separated iterator
+ */
+function* sepIter<A>(xs: Iterable<A>): IterableIterator<[boolean, A]> {
+  let comma = false;
+  for (const item of xs) {
+    yield [comma, item];
+    comma = true;
+  }
+}
+
+/**
+ * Object.entries() but skipping undefined values
+ */
+function* definedEntries<A extends object>(xs: A): IterableIterator<[string, any]> {
+  for (const [key, value] of Object.entries(xs)) {
+    if (value !== undefined) {
+      yield [key, value];
+    }
+  }
+}
+
+/**
+ * Quote string literals inside an intrinsic
```

Comment:
I still feel I don't understand why this is needed. Maybe some examples? Why isn't it possible to just use `JSON.stringify()` of the intrinsic ?

Reply:
I will add more realistic tests.

Reply:
Oh here is one: https://github.com/aws/aws-cdk/blob/huijbers/refactor-tojson/packages/@aws-cdk/core/test/cloudformation-json.test.ts/#L183

Reply:
Your intuition was correct though. I missed some edge cases. After thinking about it some more, just a straight up recursion over all strings should also do it. Added more tests to confirm.

Reply:
👍